# # 08 / 04 / 2025
# # finally : it is executed always , it is used to clen resources 

# try:
#     print("try block")
#     print(10/0)
# except ZeroDivisionError as msg :
#     print(msg)
# finally: print('finally  block')


# #  try ke sath ya to except hota hai ya finaly hota hai try labi akela nahi hota hai imp for interview 
# try: pass
# except: pass
# else:pass
# finally : pass



# try : pass
# finally : pass


# # invalid 
# # try : pass
# # else : pass 

# try : pass 



# try:
#     n1 = int(input("enter number 1: "))
#     n2 = int(input("enter number 2: "))
#     try : 
#         print(n1/n2)
#     except ZeroDivisionError :
#         print("cannot divide by zero")
# except Exception:
#     print("invalid value ")


